api/views.py

```diff
@@ -32,66 +73,94 @@ def default(self, obj):
-                (Q(visibility=Post.FOAF, author__requester__requester=request.user.id, author_id=author_id)| Q(visibility=Post.friend, author=request.user, author_id=author_id)) |
-                (Q(visibility=Post.FOAF, author__requester__accepter=request.user.id, author_id=author_id)| Q(visibility=Post.friend, author=request.user, author_id=author_id)) |
-                (Q(visibility=Post.FOAF, author__accepter__requester=request.user.id, author_id=author_id)| Q(visibility=Post.friend, author=request.user, author_id=author_id)) |
-                (Q(visibility=Post.FOAF, author__accepter__accepter=request.user.id, author_id=author_id)| Q(visibility=Post.friend, author=request.user, author_id=author_id)) )
-
-        posts = Post.objects.filter(query)
-        print posts.query
-        posts = serializers.serialize("json", posts)
-
-        #TODO Add Pagination
-        data = {"posts":posts};
-
-        return JsonResponse(data)
-    else:
-        return HttpResponse(status_code=406)
+    query = get_post_query(request)
+
+    # user specified a specific user id they want to find
+    if author_id is not None and author_id != request.user.id:
+        query = ( query ) & Q(author_id=author_id)
+    elif author_id is not None:
+        # author id = same user
+        query = Q(author_id=request.user.id)
+
+    # Check if user is valid
+    # TODO Decorator?
+    if author_id is not None:
+        try:
+            User.objects.get(id=author_id)
+        except User.DoesNotExist as e:
+            # return a better error for missing user
+            response = JsonResponse({"message":"Author with id %d does not exist" % author_id})
+            response.status_code = 404
+            return response
+
+    posts_query = Post.objects.filter(query)
+    posts = post_list(posts_query)
+
+    #TODO Add Pagination
+    data = {"posts":posts};
+
+    return JsonResponse(data)
 
+@login_required
+@require_http_methods(["GET"])
+@require_http_accept(['application/json'])
 def get_post(request,post_id=None,page="0"):
+    return_data = list()
     if post_id is not None:
-        return Post.objects.get(post_id)
+
+        query = get_post_query(request) & Q(id=post_id)
+        posts_query = Post.objects.filter(query)
+        return_data = post_list(posts_query)
+
+        if len(posts_query) == 0:
+            response = JsonResponse({"message":"Post with id %d does not exist" % post_id})
+            response.status_code = 404
+            return response
+
     else:
-        # TODO: Import these enum visibilities
-        # I don't like hardcoding...
-        return Post.objects.filter(visibility=5).order_by('-date')
+        posts_query = Post.objects.filter(visibility=Post.public).order_by('-date')
+        return_data = post_list(posts_query)
+
+    # TODO Add pagination
+    return JsonResponse({"posts":return_data})
 
-def friend_request(request,page="1"):
```

Comment:
Any ideas how we should implement this and the other methods that return none right now.